core/trino-main/src/main/java/io/trino/metadata/SystemSecurityMetadata.java

```diff
@@ -240,4 +243,19 @@ default void validateEntityKindAndPrivileges(Session session, String entityKind,
      * to be fully qualified, i.e., if the entity is a table, the name is of size three.
      */
     void setEntityOwner(Session session, EntityKindAndName entityKindAndName, TrinoPrincipal principal);
+
+    /**
+     * Returns all tables authorization information
+     */
+    Set<TableAuthorization> getAllTablesAuthorizationInfo(Session session, QualifiedTablePrefix prefix);
+
+    /**
+     * Returns all functions authorization information
+     */
+    Set<FunctionAuthorization> getAllFunctionsAuthorizationInfo(Session session, QualifiedTablePrefix prefix);
+
+    /**
+     * Returns all schemas authorization information
+     */
+    Set<SchemaAuthorization> getAllSchemasAuthorizationInfo(Session session, QualifiedTablePrefix prefix);
```

Comment:
It is odd to pass `QualifiedTablePrefix` as it is not a table. I mean if one passed `schema_name` parameter then you should return just that schema.

Reply:
when schema name is passed then it is easy, but this is an optimization for situation when only catalog is known, instead of returning all schemas from all catalogs when can just get schemas for a specific catalog